### PR TITLE
Master website event track fix method not allowed mub

### PR DESCRIPTION
**Pull request by mub-odoo**
PURPOSE

Currently, we found two same issue while
user reload the page and it return
"Method not Allowed".

   issues :-
   1) if user submitting the track proposal
       and reload it again.
   2) in event_track after event register
       and reload the page, it gave the same 
       error message "Method not Allowed".

SPECIFICATION

in this commit we are fix this problem by
redirect the user on different URL with
necessary data and in other controller
fetch that data so if user reload the page
than it always render the page instead of
giving error "Method not Allowed".

Task Id: 2500573
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_event/controllers/main.py [hunk 1]
```diff
@@ -389,8 +389,20 @@ def registration_confirm(self, event, **post):
         registrations = self._process_attendees_form(event, post)
         attendees_sudo = self._create_attendees_from_registration_post(event, registrations)
 
+        str_attendees_sudo_ids = ",".join(map(str, attendees_sudo.ids))
+        return request.redirect('/event/%s/registration/success?registration_ids=%s' % (event.id, str_attendees_sudo_ids))
```
Comment (awa-odoo) [inline addons/website_event/controllers/main.py:393]: Sugg:
`return request.redirect('/event/%s/registration/success?registration_ids=%s' % (event.id, ','.join(attendees_sudo.ids)))`

File: addons/website_event/controllers/main.py [hunk 2]
```diff
@@ -389,8 +389,20 @@ def registration_confirm(self, event, **post):
...
+        # fetch the related registrations, make sure they belong to the correct visitor / event pair
+        visitor = request.env['website.visitor']._get_visitor_from_request()
+        attendees_sudo = request.env['event.registration'].sudo().search([
+            ('id', 'in', [int(registration_id) for registration_id in registration_ids.split(',')]),
+            ('event_id', '=', event.id),
+            ('visitor_id', '=', visitor.id),
+        ])
         return request.render("website_event.registration_complete",
-            self._get_registration_confirm_values(event, attendees_sudo))
+                              self._get_registration_confirm_values(event, attendees_sudo))
```
Comment (awa-odoo) [inline addons/website_event/controllers/main.py:405]: Seems like an unnecessary change.

**Review by awa-odoo — CHANGES_REQUESTED** [review]
Looks globally ok, I left a couple comments.

Can you check the commit message?
It should look more like your PR description and contain both use cases that are fixed.
They are also some typos in it.

Thanks!

File: addons/website_event/controllers/main.py (hunk 1 above)
Comment (mub-odoo) [reply addons/website_event/controllers/main.py]: Hello @awa-odoo 
Changes Done!

File: addons/website_event/controllers/main.py (hunk 2 above)
Comment (mub-odoo) [reply addons/website_event/controllers/main.py]: Changes Done!

File: addons/website_event/controllers/main.py [hunk 3]
```diff
@@ -389,6 +389,17 @@ def registration_confirm(self, event, **post):
         registrations = self._process_attendees_form(event, post)
         attendees_sudo = self._create_attendees_from_registration_post(event, registrations)
 
+        return request.redirect('/event/%s/registration/success?registration_ids=%s' % (event.id, ",".join(map(str, attendees_sudo.ids))))
```
Comment (awa-odoo) [inline addons/website_event/controllers/main.py:392]: Ok I see why you use map, it won't join integers.

Then maybe something like `",".join([str(id) for id in attendees_sudo.ids])` ?
I would like to avoid the "map" usage and use a simple list comprehension instead.

Also, did you fix the comme message? It looks unchanged (see my previous review message).

Comment (mub-odoo) [reply addons/website_event/controllers/main.py]: Changes Done!

File: addons/website_event/controllers/main.py [hunk 4]
```diff
@@ -389,6 +389,17 @@ def registration_confirm(self, event, **post):
...
         attendees_sudo = self._create_attendees_from_registration_post(event, registrations)
 
+        return request.redirect('/event/%s/registration/success?registration_ids=%s' % (event.id, ",".join([str(id) for id in attendees_sudo.ids])))
+
+    @http.route(['/event/<model("event.event"):event>/registration/success'], type='http', auth="public", methods=['GET'], website=True, sitemap=False)
+    def event_registration_success(self, event, registration_ids):
+        # fetch the related registrations, make sure they belong to the correct visitor / event pair
+        visitor = request.env['website.visitor']._get_visitor_from_request()
+        attendees_sudo = request.env['event.registration'].sudo().search([
+            ('id', 'in', [int(registration_id) for registration_id in registration_ids.split(',')]),
```
Comment (tde-banana-odoo) [inline addons/website_event/controllers/main.py:399]: Maybe to update according to previous comment

File: addons/website_event/controllers/main.py [hunk 5]
```diff
@@ -389,6 +389,17 @@ def registration_confirm(self, event, **post):
         registrations = self._process_attendees_form(event, post)
         attendees_sudo = self._create_attendees_from_registration_post(event, registrations)
 
+        return request.redirect('/event/%s/registration/success?registration_ids=%s' % (event.id, ",".join([str(id) for id in attendees_sudo.ids])))
+
+    @http.route(['/event/<model("event.event"):event>/registration/success'], type='http', auth="public", methods=['GET'], website=True, sitemap=False)
+    def event_registration_success(self, event, registration_ids):
+        # fetch the related registrations, make sure they belong to the correct visitor / event pair
+        visitor = request.env['website.visitor']._get_visitor_from_request()
```
Comment (tde-banana-odoo) [inline addons/website_event/controllers/main.py:397]: People may try to call it from outside of expected flow. I would probably do something if we find no visitor.

**Review by tde-banana-odoo — CHANGES_REQUESTED** [review]
Review! Some ACLS / leaks concerns, we should probably strengthen some search / checks. 

Thanks !

**Comment by tde-banana-odoo** [comment]
Thanks for updating !

@robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo retry

**Comment by tde-banana-odoo** [comment]
@robodoo retry
